tdwii_plus_examples/rtbdi_creator/rtbdi_factory.py

```diff
@@ -254,13 +268,45 @@ def create_ups_from_plan_and_bdi(
     ups_ds.ScheduledWorkitemCodeSequence = scheduled_work_item_code_sequence
     plan_reference_item = _create_referenced_instances_and_access_item(plan, retrieve_ae_title)
     bdi_reference_item = _create_referenced_instances_and_access_item(bdi, retrieve_ae_title)
+
+    # because the Reference Patient Photo Sequence already contains Study and Series UID information... it isn't
+    # necessary to load the patient photo to get that information.
+    patient_photo_reference_items = []
+    if enable_photo_ref and "ReferencedPatientPhotoSequence" in plan:
+        for patient_photo_seq_item in plan.ReferencedPatientPhotoSequence:
+            patient_photo_reference_item = _create_referenced_instances_and_access_item(
+                patient_photo_seq_item, retrieve_ae_title
+            )
+            patient_photo_reference_items.append(patient_photo_reference_item)
+
+    # This makes the assumption that the Study and Series Instance UIDs for the Setup images are the same as the plan
```

Comment:
with different modalities for the RT Plan and the RT Images, this can actually never be right.

Reply:
Excellent point, re Series Instance UID.
That makes for an interesting issue for referenced objects that only have instance level references... The scheduler needs to retrieve the object using a relational query so it can construct the navigational reference.
I'll back out the setup image part for now and address those and RT Image references in a separate PR.

Thank you for the useful feedback.

Reply:
You are welcome. But I am not aware that any archive has implemented relational queries. So, how is this supposed to work in general for UPSs when there is only the Instance Information?
For most of the composite IODs there is the Common Instance Reference Module that provides the hierarchical information for the plain instance references in an instance...